source/common/router/config_impl.h

```diff
@@ -196,12 +197,25 @@ class RouteEntryImplBase : public RouteEntry, public Matchable, public RedirectE
   std::string newPath(const Http::HeaderMap& headers) const override;
 
   // Router::Matchable
-  bool matches(const Http::HeaderMap& headers, uint64_t random_value) const override;
+  const Route* matches(const Http::HeaderMap& headers, uint64_t random_value) const override;
 
   // Router::Route
   const RedirectEntry* redirectEntry() const override;
   const RouteEntry* routeEntry() const override;
 
+  // Throws an error if any of the clusters held by this Route are
```

Comment:
1) Method comments use /** */ syntax.
2) move method definitions above interface overrides (above 199(
3) Remove clunky/alternative comment, this is fine.